src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -955,6 +962,51 @@ public function getPlusType(Expr $left, Expr $right, callable $getTypeCallback):
 			]);
 		}
 
+		if (
+			($leftIsArray->yes() && $rightIsArray->no())
+			|| ($leftIsArray->no() && $rightIsArray->yes())
+		) {
+			return new ErrorType();
+		}
+
+		if (
+			($leftIsArray->yes() && $rightIsArray->maybe())
+			|| ($leftIsArray->maybe() && $rightIsArray->yes())
+		) {
+			if ($leftType instanceof SubtractableType && $leftType->getSubtractedType() !== null) {
```

Comment:
This logic is wrong. Instead, `MixedType::isArray()` has to be implemented correctly in regard to subtracted type.

You can actually fix all the `is*` methods there (in a separate PR) because they don't regard what subtracted yet.

Reply:
Similar to what I had in MixedType in https://github.com/phpstan/phpstan-src/pull/1579/commits/56687c73429185895c4d520d31ef734d783e83ff ?

Reply:
Yeah, but `$this->subtractedType->isArray()->yes()` is still wrong. If you subtract `array{}`, it can still be a different array.

You need to ask something like `$this->subtractedType->isSuperTypeOf(new ArrayType(new MixedType(), new MixedType()))->yes()`, that covers all arrays.

Reply:
here we go https://github.com/phpstan/phpstan-src/pull/1656

Reply:
Any `instanceof SubtractableType` should no longer be necessary here.

Reply:
fixed and rebased